newsplease/crawler/commoncrawl_crawler.py

```diff
@@ -216,7 +216,8 @@ def __start_commoncrawl_extractor(warc_download_url, callback_on_article_extract
                                   log_level=logging.ERROR,
                                   delete_warc_after_extraction=True,
                                   continue_process=True,
-                                  log_pathname_fully_extracted_warcs=None):
+                                  log_pathname_fully_extracted_warcs=None,
+                                  extractor_cls=CommonCrawlExtractor):
```

Comment:
Why do you think passing the CommonCrawlExtractor class here is necessary or what are the advantages of doing so compared to how it's currently implemented? Other than that, the PR is fine to me, i.e., making the filter method public.

Reply:
The idea is that a user can subclass `CommonCrawlExtractor` themselves and then pass their subclass to `crawl_from_commoncrawl`. This way they can provide programmatic filters by overriding `filter_record(...)`.

Reply:
This makes sense but I think this requires some brief explanation, e.g., as a comment perhaps at the following points: https://github.com/fhamborg/news-please/blob/563de2b7d7058cf89b894d581fea023a82b3f21a/newsplease/crawler/commoncrawl_crawler.py#L277 and here https://github.com/fhamborg/news-please/blob/563de2b7d7058cf89b894d581fea023a82b3f21a/newsplease/examples/commoncrawl.py